packages/integration/src/azure/config.ts

```diff
@@ -68,44 +91,103 @@ export type AzureClientSecretCredential = {
 };
 
 /**
- * Authenticate using a managed identity available at the deployment environment.
+ * A managed identity credential.
  * @public
  */
-export type AzureManagedIdentityCredential = {
+export type AzureManagedIdentityCredential = AzureCredentialBase & {
   /**
    * The clientId
    */
   clientId: string;
 };
 
 /**
- * Credential used to authenticate to Azure Active Directory.
+ * A personal access token credential.
  * @public
  */
-export type AzureCredential =
+export type PersonalAccessTokenCredential = AzureCredentialBase & {
+  token: string;
+};
+
+/**
+ * The general shape of a credential that can be used to authenticate to Azure DevOps.
+ * @public
+ */
+export type AzureDevOpsCredentialLike = Partial<AzureClientSecretCredential> &
+  Partial<AzureManagedIdentityCredential> &
+  Partial<PersonalAccessTokenCredential>;
+
+/**
+ * Credential used to authenticate to Azure DevOps.
+ * @public
+ */
+export type AzureDevOpsCredential =
   | AzureClientSecretCredential
-  | AzureManagedIdentityCredential;
-export const isAzureClientSecretCredential = (
-  credential: Partial<AzureCredential>,
-): credential is AzureClientSecretCredential => {
-  const clientSecretCredential = credential as AzureClientSecretCredential;
+  | AzureManagedIdentityCredential
+  | PersonalAccessTokenCredential;
 
+/**
+ * Verifies that the given credential is an AzureClientSecretCredential.
+ * @param credential - The credential to verify
+ * @public
+ */
+export function isAzureClientSecretCredential(
+  credential: AzureDevOpsCredentialLike,
+): credential is AzureClientSecretCredential {
```

Comment:
This took me a while, but I got it working in the end :)

Reply:
@Rugvip I refactored `AzureDevOpsCredential` into a discriminated type union, but I wanted to keep the `kind` out of the configuration since that would make the configuration more verbose. But that means I must map an object similar to a credential (`AzureDevOpsCredentialLike`) to an object that satisfies the `AzureDevOpsCredential` type union. Even if I put `kind` in the config, I'd still have to validate that all required properties are present. 

So I added a mapping of credential kinds and fields and the asAzureDevOpsCredential` method in `config.ts` to perform the validation and mapping. 

I'm still trying to decide whether a more explicit and verbose implementation, for example, using specific functions to validate and map each credential kind, is better. I'm also trying to decide if it's better to put `kind` in the config. 

I would appreciate your opinion on this!